File: frontend/src/microservices/tour-group/objects/TourGroupMembershipResponse.ts
// 本文件定义 tour-group 模块的 `TourGroupMembershipResponse`，作为响应数据并提供 JSON 编解码。

export type TourGroupMembershipResponse = {
  membershipId: string
  userId: string
  userDisplayName?: string
  status: string
  joinedAt: string
}
export const tourGroupMembershipResponseFromJson = (json: string): TourGroupMembershipResponse =>
  JSON.parse(json) as TourGroupMembershipResponse

export const tourGroupMembershipResponseToJson = (value: TourGroupMembershipResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/tour-group/objects/TourGroupMembershipResponse.scala
// TourGroupPlannerModels 定义团体游模块的请求和响应模型。

package com.typesafe.travel.tourgroup.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TourGroupMembershipResponse(membershipId: String, userId: String, userDisplayName: String, status: String, joinedAt: String)
object TourGroupMembershipResponse:
  given sourceEncoder: Encoder[TourGroupMembershipResponse] = deriveEncoder
  given sourceDecoder: Decoder[TourGroupMembershipResponse] = deriveDecoder
